### PR TITLE
Close #2946

### LINKED ISSUE #2946: Boundary Object Update for Icepak Source
### Before submitting the issue

- [X] I have searched among the existing issues
- [X] I am using a Python virtual environment

### Description of the bug

Unable to change the source power value of Icepak Source by updating the dictionary. The boundary dictionary update is not calling Icepak source update function but rather EditTerminal function.

 

pyaedt ERROR: Error in :   File "C:\Users\skrishna\AppData\Roaming\pyaedt_env_ide\v232\lib\site-packages\pyaedt\modules\Boundary.py", line 630, in update
pyaedt ERROR: GetModule.EditTerminal - update -  ATTRIBUTE ERROR.
pyaedt ERROR: Arguments with values: 
pyaedt ERROR: Check Online documentation on: https://aedt.docs.pyansys.com/search.html?q=update
pyaedt WARNING: Update of Total Power Failed. Check needed arguments
pyaedt ERROR: Error in :   File "C:\Users\skrishna\AppData\Roaming\pyaedt_env_ide\v232\lib\site-packages\pyaedt\modules\Boundary.py", line 630, in update
pyaedt ERROR: GetModule.EditTerminal - update -  ATTRIBUTE ERROR.
pyaedt ERROR: Arguments with values: 
pyaedt ERROR: Check Online documentation on: https://aedt.docs.pyansys.com/search.html?q=update

### Steps To Reproduce

1. Open an existing Icepak project with Source Object in Pyaedt
2. Change the total power of Icepak source by updating the boundary object
`ipk.boundaries[0]['Total` Power'] = "2W"

### Which Operating System are you using?

Windows

### Which Python version are you using?

3.7

### Installed packages

-

### PR DESCRIPTION
**Pull request by sivasu37**
In Icepak, a source boundary can be created by selecting a face and assigning the power. It sets the boundary type as "Source". This boundary type is clashing with the Maxwell Source type. The fix will modify the boundary type as "IcepakSource". It is done by checking if the design type is Icepak

    @property
    def type(self):
        if self._app.design_type == "Icepak" and self._type == "Source":
            return "SourceIcepak"
        else:
            return self._type

### CODE REVIEW

**Comment by lorenzovecchietti** [comment]
The old power_budget function would work just with sources created from the gui/loaded from file. Now it is looking for "SouceIcepak" instead of "Source".
Hopefully, this fix the failing unit test.

File: pyaedt/modules/Boundary.py [hunk 1]
```diff
@@ -317,10 +317,21 @@ def __init__(self, app, name, props, boundarytype):
         self._app = app
         self._name = name
         self.props = BoundaryProps(self, OrderedDict(props))
-        self.type = boundarytype
+        self._type = boundarytype
         self._boundary_name = self.name
         self.auto_update = True
 
+    @property
+    def type(self):
```
Comment (MaxJPRey) [inline pyaedt/modules/Boundary.py:325]: The property is missing a docstring which is causing the documentation build to fail.
As soon as it is added, the workflow should pass entirely.

Comment (MaxJPRey) [inline pyaedt/modules/Boundary.py:325]: The property is missing a docstring which is causing the documentation build to fail.
As soon as it is added, the workflow should pass entirely.
```suggestion
    def type(self):
        """Boundary type."""
```

File: pyaedt/modules/Boundary.py [hunk 2]
```diff
@@ -317,10 +317,21 @@ def __init__(self, app, name, props, boundarytype):
...
+        if self._app.design_type == "Icepak" and self._type == "Source":
+            return "SourceIcepak"
+        else:
+            return self._type
+
+    @type.setter
+    def type(self, type_in):
+        self._type = type_in
```
Comment (MaxJPRey) [inline pyaedt/modules/Boundary.py:333]:
```suggestion
    def type(self, value):
        self._type = value
```

**Review by lorenzovecchietti — APPROVED** [review]

**Pull request merged**